src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/BuildIntegratedProjectSystem.cs

```diff
@@ -150,13 +157,17 @@ private IScriptExecutor ScriptExecutor
                             var childName =
                                 await EnvDTEProjectUtility.GetCustomUniqueNameAsync(childReference.SourceProject);
 
-                            childReferences.Add(childName);
-
-                            // avoid looping by checking if we already have this project
-                            if (!uniqueProjects.Contains(childName))
+                            // Skip projects which have ReferenceOutputAssembly=false
+                            if (!excludedProjects.Contains(childName, StringComparer.Ordinal))
```

Comment:
It should always be the same case due to the way NuGet generates the unique name, but I'll switch this to case-insensitive to be safe.